### DISCUSSION TITLE
HTML code

### DISCUSSION THREAD

**Discussion by GLEBKA-HLEBKA** [community/community#130949] [Programming Help]
### Body

Sry for my English first!!!
I have a HTML code which show me a DB where information is stored
I need an action when if I click on Button "Открыть"  (Open) another html page was opening for me and give me an information unique by ID
For example, for 1st ID  unique table opens with 
unique info for the 1st ID
How can I do this?
![Скриншот-04-07-2024 03_08_54](https://github.com/community/community/assets/174010820/098b3853-39dd-4270-bcb5-af7c757a800b)

### Guidelines

- [x] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by SanjaySRocks** [comment]
You should paste your code as a code.

**Comment by localnetwork** [comment]
HTML does not inherently support dynamic routes like http://your-website-url.com/person/john-doe.html. However, you can achieve similar functionality using query parameters, such as http://your-website-url.com/person?id=john-doe, and dynamically loading data based on those parameters.

To implement this, you can store the information in a JSON file. For example:

data/people.json:

[ { "name": "Person 1", "id": "person-1", "job_title": "Supervisor" }, { "name": "Person 2", "id": "person-2", "job_title": "Founder" } ]

Folder Structure:

/your-project
│
├── /data
│ └── people.json
│
├── /person
│ └── index.html
│
└── script.js

Fetching Data with Query Parameters
In your JavaScript, you can use fetch to load the data and query the specific person using URLSearchParams. Here’s an example:

/person/index.html:

`<!DOCTYPE html> <html lang="en"> <head> <meta charset="UTF-8"> <meta name="viewport" content="width=device-width, initial-scale=1.0"> <title>Person Details</title> </head> <body> <h1 id="name"></h1> <p id="job-title"></p>

<script src="../script.js"></script>  
</body> </html>`


**script.js:**

document.addEventListener('DOMContentLoaded', async () => {
// Get the query parameter from the URL
const urlParams = new URLSearchParams(window.location.search);
const personId = urlParams.get('id');

`// Fetch the JSON data  
const response = await fetch('/data/people.json');  
const people = await response.json();  

// Find the person by ID  
const person = people.find(p => p.id === personId);  

if (person) {  
    // Display the person's details  
    document.getElementById('name').textContent = person.name;  
    document.getElementById('job-title').textContent = person.job_title;  
} else {  
    document.getElementById('name').textContent = 'Person not found';  
} });`

In this example, when visiting http://your-website-url.com/person?id=person-1, the JavaScript will extract the id from the URL and use it to fetch and display the relevant person’s details from the JSON file.
